src/profile-logic/symbol-store.js

```diff
@@ -4,139 +4,231 @@
-   *                                needs to return a promise of [addr, syms] (the symbol table).
-   */
   constructor(dbNamePrefix: string, symbolProvider: SymbolProvider) {
     this._symbolProvider = symbolProvider;
     this._db = new SymbolStoreDB(`${dbNamePrefix}-symbol-tables`);
-
-    // A set of strings identifying libraries that we have requested
-    // symbols for but gotten an error back.
-    this._failedRequests = new Set();
-
-    // A map with one entry for each library that we have requested (but not yet
-    // received) symbols for. The keys are strings (libid), and the values are
-    // promises that resolve to [ addrs, index, buffer ] symbol tables.
-    this._requestedSymbolTables = new Map();
   }
 
-  _getSymbolTable(lib: Library): Promise<SymbolTableAsTuple> {
-    const { debugName, breakpadId } = lib;
-    const libid = `${debugName}/${breakpadId}`;
-
-    if (this._failedRequests.has(libid)) {
-      return Promise.reject(
-        new Error(
-          "We've tried to request a symbol table for this library before and failed, so we're not trying again."
-        )
-      );
-    }
-
-    const existingRequest = this._requestedSymbolTables.get(libid);
-    if (existingRequest !== undefined) {
-      // We've already requested a symbol table for this library and are
-      // waiting for the result, so just return the promise for the existing
-      // request.
-      return existingRequest;
-    }
-
-    // Try to get the symbol table from the database
-    const symbolTablePromise = this._db
-      .getSymbolTable(debugName, breakpadId)
-      .catch(() => {
-        // Request the symbol table from the symbol provider.
-        const symbolTablePromise = this._symbolProvider
-          .requestSymbolTable(debugName, breakpadId)
-          .catch(error => {
-            console.error(`Failed to symbolicate library ${debugName}`, error);
-            this._failedRequests.add(libid);
-            this._requestedSymbolTables.delete(libid);
-            throw error;
-          });
-
-        // Once the symbol table comes in, store it in the database, but don't
-        // let that block the promise that we return to our caller.
-        symbolTablePromise.then(symbolTable => {
-          this._db.storeSymbolTable(debugName, breakpadId, symbolTable).then(
-            () => {
-              this._requestedSymbolTables.delete(libid);
-            },
-            error => {
-              console.error(
-                `Failed to store the symbol table for ${debugName} ${breakpadId} in the database:`,
-                error
-              );
-              // We'll keep the symbolTablePromise in _requestedSymbolTables so
-              // that we'll the symbolTable around for future requests even though
-              // we failed to put it into the database.
-            }
-          );
-        });
-
-        return symbolTablePromise;
+  _storeSymbolTableInDB(
```

Comment:
can you please mention somewhere that only the symbols that are dumped from the binary (not the symbols coming from the server) will be stored ?